lib/install.js

```diff
@@ -220,12 +213,12 @@ function install (gyp, argv, callback) {
       // download the tarball, gunzip and extract!
 
       if (tarPath) {
-        var input = fs.createReadStream(tarballUrl)
+        var input = fs.createReadStream(tarPath)
         input.pipe(gunzip).pipe(extracter)
         return
       }
 
-      var req = download(tarballUrl)
+      var req = download(release.tarballUrl)
       if (!req) return
```

Comment:
old request bug I believe, related to https://github.com/nodejs/node-gyp/issues/114, I dare not touch it without tests to understand it